lib/alloc.js

```diff
@@ -0,0 +1,40 @@
+class CodeAllocator {
+  constructor (sliceSize) {
+    this.sliceSize = sliceSize;
+    this.slicesPerPage = Process.pageSize / sliceSize;
+
+    this.pages = [];
+    this.free = [];
+  }
+
+  allocateSlice () {
+    let slice = this.free.pop();
+    if (slice !== undefined) {
+      return slice;
+    }
+
+    return this._allocatePage();
+  }
+
+  _allocatePage () {
+    const page = Memory.alloc(Process.pageSize);
+    for (let i = 1; i < this.slicesPerPage; i++) {
```

Comment:
Is it intentional that the first slice isn't used? The reason we do this in Gum's code allocator is that we use that space for a header, but it doesn't look like we do that here.

Reply:
it adds all slices to the free list except for the first one as it's being allocated for use (we could give a different name to the method if that makes it clearer).

Reply:
Ahh, makes sense. Thanks for clarifying!